workflow/util/util.go

```diff
@@ -244,7 +246,21 @@ func SubmitWorkflow(wfIf v1alpha1.WorkflowInterface, wf *wfv1.Workflow, opts *Su
 	if err != nil {
 		return nil, err
 	}
-	return wfIf.Create(wf)
+	if opts.DryRun {
```

Comment:
Will this action do the real POST thing , the workflow will be created in the code

Reply:
I followed the [API documentation](https://kubernetes.io/docs/reference/using-api/api-concepts/#dry-run) and added the ```dryRun=All``` parameter which will run checks to make sure the request is valid, mutates the request, merges patches, defaults field values, and validates the schema validation. It shouldn't create anything.

All of that will be done on the server side and is only possible, by default, on kubernetes versions >= 1.13 ( has to be enabled for version 1.12 )